hawkular-bus-common/src/main/java/org/hawkular/bus/common/consumer/BytesMessageInputStream.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.bus.common.consumer;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.jms.BytesMessage;
+import javax.jms.JMSException;
+
+/**
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ *
+ */
+public class BytesMessageInputStream extends InputStream {
```

Comment:
I think you need to override more methods in there. close() for example - when this stream is closed, don't you want to close the delegate?